pallets/clusters/src/lib.rs

```diff
@@ -31,9 +31,18 @@ mod benchmarking;
 /// The **settings** of a **Role**
 #[derive(Encode, Decode, Clone, PartialEq, Debug, Eq, scale_info::TypeInfo)]
 pub struct RoleSetting {
-	/// Name of the setting
+	/// Index of the name of the role setting to be stored
 	pub name: Compact<u64>,
-	/// The data associated with the Role
+	/// The data associated with the Role to be stored
+	pub data: Vec<u8>,
+}
+
+/// The **settings** of a **Role**
+#[derive(Encode, Decode, Clone, PartialEq, Debug, Eq, scale_info::TypeInfo)]
+pub struct Setting {
```

Comment:
Can we call this `Setting` and the current `RoleSetting` `CompactSetting` or something to avoid confusion?
Right know it's very confusing.